### PR TITLE
RavenDB-13838 WhatChanged on document metadata return wrong results

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.Client/Json/BlittableOperation.cs [hunk 1]
```diff
@@ -56,6 +56,11 @@ public static bool EntityChanged(BlittableJsonReaderObject newObj, DocumentInfo
             {
                 if (changes == null)
                     return true;
+                if (field.Equals(LastModified) ||
+                    field.Equals(Collection) ||
```
Comment (ayende) [inline src/Raven.Client/Json/BlittableOperation.cs]: Why are we ignoring the collection field or the id here?

Comment (ppekrol) [reply src/Raven.Client/Json/BlittableOperation.cs]: We already do that 10 lines below for new fields.

Comment (efratshenhar) [reply src/Raven.Client/Json/BlittableOperation.cs]: Ccollection doesn't need to be there but we are removing the id in JsonConverter.ToBlittable and we don't want to mark that as removed

Comment (ayende) [reply src/Raven.Client/Json/BlittableOperation.cs]: ok

**Pull request merged**